frontend/src/components/Registration.js

```diff
@@ -0,0 +1,237 @@
+import React, { useState } from "react";
```

Comment:
Ez a file rettenetesen hosszú, majdnem 250 sor, mindenképpen érdemes lenne refactorálni, pár dolgot más file-ba kiszervezni, ötletek:
- hook-os logikákat egyedi hookba kiszervezni
- schema validálás külön fileba. értem, hogy problémásak a hibaüzenetek, mert nincs t. lehet nem is kell, elég csak a resource stringet odaadni a Yup-nak és a kiírásnál abból fordítani: https://stackoverflow.com/questions/49389202/yup-doesnt-work-properly-with-i18n
- a React elemeken valahogy egyszerűsíteni, esetleg valamelyik részt kiszervezni külön komponensbe (például az alján a gomb és a modal, vagy a tax number)